Sources/MongoSwift/Operations/Operation.swift

```diff
@@ -3,11 +3,34 @@
 internal protocol Operation {
     /// The result type this operation returns.
     associatedtype OperationResult
+    /// Indicates how this operation interactions with `Connection`s.
+    var connectionUsage: ConnectionUsage { get }
+
     /// Executes this operation using the provided connection and optional session, and returns its corresponding
     /// result type.
     func execute(using connection: Connection, session: ClientSession?) throws -> OperationResult
 }
 
+extension Operation {
+    /// This is the behavior of most operations so default to this.
+    internal var connectionUsage: ConnectionUsage { return .uses }
+}
+
+/// Uses to indicate how an `Operation` type uses `Connection`s passed to its execute method.
+internal enum ConnectionUsage {
```

Comment:
nitpick: I think it might be more consistent to call this a "strategy".

What is the difference between an operation which retains a connection (owns), and one which "steals" a connection? It seems like in both cases we're saying "the connection passed into this operation should become the responsibility of the operation, it should close it/return it eventually" It seems to me the distinction lies in the initialization of the operation. 

Rather, would it make sense to have only two strategies: `retained(conn)` and `unretained`, and the stealing would be done though `init(stealing: conn, session)`?

Reply:
strategy definitely sounds better, i didn't really like this name.

I agree that these cases are very similar. I was mainly trying to avoid calling the
`try session?.getConnection(forUseWith: client) ?? client.connectionPool.checkOut()` line every time we construct an operation like this. but at the moment anyway that is only 3 places (once for each of collection/db/client watch) so it seems acceptable.

that said I feel like we can do better on naming the cases, because it's still not quite true that e.g. `NextOperation` owns the connection or takes responsibility for it.... Rather, the cursor / change stream that created the operation owns it. in the `WatchOperation` case, the operation temporarily owns it, but quickly passes off ownership to the `ChangeStream`.

maybe `.bound(to: Connection)` and `.unbound` better conveys this? i.e. based on the context that the operation was created it in we've already bound it to a particular connection.

Reply:
that sgtm, “retained” immediately came to mine,  but I think bound similarly conveys intent. Perhaps a helper function is in order for the repetitive code issue, I thought at some point we discussed `resolveConnection(session, pool)`?

Reply:
yeah, i feel like we had that in a previous PR but ended up not needing it? seems useful now though and we can also use it in `MongoCursor`'s init which does the same thing.

I also added a `releaseConnection` method that handles the reverse logic of releasing a connection that was obtained via this method.